### PR TITLE
Move all schemas to a single file schemas.py

### PR DESCRIPTION
**Pull request by vdbergh**
This reduces clutter and avoids duplicate definitions.

Also:

The run schema has been tightened.

Various tweaks in other schemas.

Requires upgrading the vtjson package.

Note that this PR contains #1921 , which creates a small conflict if #1921 is committed first.

### CODE REVIEW

**Comment by vdbergh** [comment]
Drafted since I forgot about the workerdb schema.

**Comment by vdbergh** [comment]
Undrafted.

**Comment by vdbergh** [comment]
Hmm, seems like a small bug in fishtest... 
```
net (value:{'_id': ObjectId('65e1246271ab4fc7757e48ad'), 'user': 'user00', 'name': 'nn-baff1ede1f90.nnue', 'do...[TRUNCATED]...}) is not of type 'first_test_before_last': can't compare offset-naive and offset-aware datetimes 
```

**Comment by vdbergh** [comment]
Actually the bug is in this PR since I changed how nns are written to the db.

**Comment by vdbergh** [comment]
No it was actually a bug in fishtest. Dates in mongodb are not timezone aware.

File: server/fishtest/rundb.py [hunk 1]
```diff
@@ -469,19 +278,37 @@ def get_run_pgns(self, run_id):
...
-    def upload_nn(self, userid, name, nn):
-        self.nndb.insert_one({"user": userid, "name": name, "downloads": 0})
-        return {}
+    def write_nn(self, net):
+        validate(nn_schema, net, "net")
+        self.nndb.replace_one({"name": net["name"]}, net, upsert=True)
+
+    def get_nn(self, name):
+        net = self.nndb.find_one({"name": name}, {"nn": 0})
+        # pymong is not timezone aware
```
Comment (ppigazzini) [inline server/fishtest/rundb.py:287]:
```diff
- # pymong is not timezone aware
+ # pymongo is not timezone aware
```

**Comment by vdbergh** [comment]
Corrected!

**Comment by ppigazzini** [comment]
DEV updated, fixed.

**Comment by ppigazzini** [comment]
Server crash uploading a net.
```yaml
usr00@dfts-0:~$ sudo journalctl -u fishtest@6544 --since "10 minutes ago"
-- Logs begin at Sat 2024-03-30 15:49:04 CET, end at Tue 2024-04-02 23:29:44 CEST. --
Apr 02 23:28:41 dfts-0 pserve[11730]: 2024-04-02 23:28:41,382 ERROR [waitress][waitress-0] Exception while serving /upload
Apr 02 23:28:41 dfts-0 pserve[11730]: Traceback (most recent call last):
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/tweens.py", line 13, in _error_handler
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = request.invoke_exception_view(exc_info)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/view.py", line 786, in invoke_exception_view
Apr 02 23:28:41 dfts-0 pserve[11730]:     raise HTTPNotFound
Apr 02 23:28:41 dfts-0 pserve[11730]: pyramid.httpexceptions.HTTPNotFound: The resource could not be found.
Apr 02 23:28:41 dfts-0 pserve[11730]: During handling of the above exception, another exception occurred:
Apr 02 23:28:41 dfts-0 pserve[11730]: Traceback (most recent call last):
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/waitress/channel.py", line 428, in service
Apr 02 23:28:41 dfts-0 pserve[11730]:     task.service()
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/waitress/task.py", line 168, in service
Apr 02 23:28:41 dfts-0 pserve[11730]:     self.execute()
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/waitress/task.py", line 436, in execute
Apr 02 23:28:41 dfts-0 pserve[11730]:     app_iter = self.channel.server.application(environ, start_response)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/waitress/proxy_headers.py", line 64, in translate_proxy_headers
Apr 02 23:28:41 dfts-0 pserve[11730]:     return app(environ, start_response)
Apr 02 23:28:41 dfts-0 pserve[11730]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/router.py", line 270, in __call__
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = self.execution_policy(environ, self)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/router.py", line 276, in default_execution_policy
Apr 02 23:28:41 dfts-0 pserve[11730]:     return router.invoke_request(request)
Apr 02 23:28:41 dfts-0 pserve[11730]:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/router.py", line 245, in invoke_request
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = handle_request(request)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/tweens.py", line 43, in excview_tween
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = _error_handler(request, exc)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/tweens.py", line 17, in _error_handler
Apr 02 23:28:41 dfts-0 pserve[11730]:     reraise(*exc_info)
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/util.py", line 733, in reraise
Apr 02 23:28:41 dfts-0 pserve[11730]:     raise value
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/tweens.py", line 41, in excview_tween
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = handler(request)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/router.py", line 143, in handle_request
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = _call_view(
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/view.py", line 674, in _call_view
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = view_callable(context, request)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/viewderivers.py", line 512, in csrf_view
Apr 02 23:28:41 dfts-0 pserve[11730]:     return view(context, request)
Apr 02 23:28:41 dfts-0 pserve[11730]:            ^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/viewderivers.py", line 427, in rendered_view
Apr 02 23:28:41 dfts-0 pserve[11730]:     result = view(context, request)
Apr 02 23:28:41 dfts-0 pserve[11730]:              ^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/env/lib/python3.12/site-packages/pyramid/viewderivers.py", line 141, in _requestonly_view
Apr 02 23:28:41 dfts-0 pserve[11730]:     response = view(request)
Apr 02 23:28:41 dfts-0 pserve[11730]:                ^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/fishtest/views.py", line 277, in upload
Apr 02 23:28:41 dfts-0 pserve[11730]:     if request.rundb.get_nn(filename):
Apr 02 23:28:41 dfts-0 pserve[11730]:        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]:   File "/home/usr00/fishtest/server/fishtest/rundb.py", line 288, in get_nn
Apr 02 23:28:41 dfts-0 pserve[11730]:     if "first_test" in net:
Apr 02 23:28:41 dfts-0 pserve[11730]:        ^^^^^^^^^^^^^^^^^^^
Apr 02 23:28:41 dfts-0 pserve[11730]: TypeError: argument of type 'NoneType' is not iterable

```

**Comment by vdbergh** [comment]
I think I fixed it.

**Comment by ppigazzini** [comment]
> I think I fixed it.

Fixed!

**Comment by ppigazzini** [comment]
PROD updated by hand to thoroughly test the PR before the merge.
Info> PROD ans DEV have been running with python 3.12.2 since few days.

**Comment by vdbergh** [comment]
Last commit is a trivial change to deal with #1923

**Comment by ppigazzini** [comment]
PROD and DEV updated.

**Review by ppigazzini — APPROVED** [review]
No new issues in the last few hours testing the PR on PROD and DEV.

**Comment by ppigazzini** [comment]
PROD updated, thank you @vdbergh :)

**Pull request merged**